### DISCUSSION TITLE
Benchmark config info incomplete

### DISCUSSION THREAD

**Discussion by jedahan** [zimfw/zimfw#2] [Ideas]
taco taco taco tacos doooooom! (Thanks for writing zim! It looks rad!)

It would be awesome, if for https://github.com/Eriner/zim/wiki/Speed you included the full configurations (.zshrcs) for each test, perhaps in a subdirectory, or separate branch. I wanna see just how zim has such a speed advantage to perhaps help improve the other projects.

Cheers

**Comment by Eriner** [comment]
Sure, I'm busy this weekend, but I can do that on Monday.

**Comment by Eriner** [comment]
That said, to be clear, the configs mentioned are the only things that I changed from the default. I made no additional changes. Now that I think about it, I think just making that clear would suffice. I guess I never explicitly mentioned that. Is that reasonable, @jedahan ?

**Comment by Eriner** [comment]
Here is the change to the wiki: https://github.com/Eriner/zim/wiki/Speed/_compare/69aab52cee412d5b3398b0786ef426e8a56f3d05...ec6c8871a478a74e4d171df33e7fe77bfd32b654

But you are right. In fact, I will create a repository with a script that will initialize and setup each framework, and test for the user, logging to a file. That seems like a worthwhile thing to do.

**Comment by jedahan** [comment]
Making that clarification is certainly appreciated, but it's even nicer that you are going to make a benchmarking script as that will be helpful for chasing down future improvements/regressions etc. Thanks!

**Comment by Eriner** [comment]
@jedahan Agreed. I will set that up on Monday. Thanks for the idea!

**Comment by Eriner** [comment]
@jedahan I am near-finished with the script. Currently, it does not take any optional command line parameters, which is what I will be adding before I publish this repo. I would also like to include thing like antigen, which I don't have added at the moment.

This is the output of running the script:

```
This may take a LONG time, as it runs each framework startup 100 times
Average startup times for each framework will be printed as the tests progress.

The average startup time for oh-my-zsh is: 1.2808599999999999
The average startup time for zplug is: 1.3274899999999994
The average startup time for prezto is: 0.87578999999999996
The average startup time for zim is: 0.40482999999999997
```

Let me know what you think! Also, it has a 'running' indicator, so it's easy to make sure that it's running.

**Comment by Eriner** [comment]
@jedahan Here is the repo: https://github.com/Eriner/zsh-framework-benchmark

I am going to close this issue now, and will continue work in that repo.
